gensrc/proto/types.proto

```diff
@@ -63,3 +67,150 @@ message PUniqueId {
     required int64 lo = 2;
 };
 
+message PGenericType {
```

Comment:
This is a same datatype define in `data.proto`, how about unify them?
I will do this in PR #7939

Reply:
That's a good idea, but I think the datatype define in `data.proto` can not describe complex types very well, `PGenericType` + `PList` + `PMap` + `PField` +  `PStruct` + `PDecimal` may be better. And BTW,  datatype should not  to complex, it should be used more easier between languages